iterators.tex

```diff
@@ -6349,8 +6356,8 @@
 \oldtxt{\tcode{begin} and \tcode{end} are required to be amortized constant time.}
 
 \pnum
-\newtxt{Given an object \tcode{t} of type \tcode{T}, \oldtxt{the type \tcode{T} models}
-\tcode{Range\newtxt{<T>()}} \newtxt{is satisfied} if and only if}
+\newtxt{Given a value \tcode{t} of type \tcode{T}, \tcode{Range<T>()} is satisfied
+if and only if}
```

Comment:
> I think it's fine if such a type satisfies `Range` but not `SizedRange`.

We had a discussion about a thousand years ago in which you said that you want a template that is declared to require a `Range` to be able to assume that the actual parameter is a `SizedRange` if it meets the syntax. That's why I originally added this requirement that forbids passing a non-`SizedRange` that looks like a `SizedRange` as a `Range`.

If you've decided that you instead want to require anything that cares about `SizedRange`-ness to explicitly declare an overload/specialization for `SizedRange` - as you did with `distance` - then we can get rid of this bullet.